src/ucp/proto/proto_common.h

```diff
@@ -122,8 +122,12 @@ typedef struct {
     /* Map of unsuitable lanes */
     ucp_lane_map_t          exclude_map;
 
-    /* Memory type that the transport should be capable of registering. */
-    ucs_memory_type_t       reg_mem_type;
+    /* Memory type that the transport should be capable of registering.
+     * In case of zcopy protocols (e.g. rndv/put/zcopy, am/egr/single/zcopy,
+     * etc) it should be the same type as user buffer, which is specified
+     * in select_param->mem_type. For mtype protocols (rndv/put/mtype and
+     * rndv/get/mtype) it represents fragment memory type. */
```

Comment:
Memory type of the buffer used data transfer on the transport level. If SEND_ZCOPY is set, it is expected to be the user buffer memory type. Alternatively, it refers to the type of memory used for bounce buffers (either in the UCP or UCT layer) where data needs to be copied as part of the protocol.